### PR TITLE
[NTV-13] Supporting deep links to a specific reply.

### PR DESCRIPTION
**Pull request by singhhari**
# 📲 What

One of the final pieces to our deep link integration with comment threading is the ability to take a user to a specific reply. We're using the replyId as a query parameter and scrolling the table view to a particular index, if the replyId matches.

# 🤔 Why

This is part of our larger goal to drive engagement with deeplinks. This should be something we incorporate into all new feature releases. A good gameplan would probably be to release the feature, add analytics support and then finish it off with a supporting deeplink.

# 🛠 How

Most of the work involving deeplinks is done in two files: `AppDelegateViewModel.swift` and `Navigation.swift`. It's in these two files that we parse the route, determine the relevant query parameters and subsequently navigate the user to the correct point in the stack.

# ✅ Acceptance criteria

I used the following links to test:
Update: https://staging.kickstarter.com/projects/fjorden/fjorden-iphone-photography-reinvented/posts/3254626/comments?comment=Q29tbWVudC0zMzU2MTY4Ng%3D%3D&reply=Q29tbWVudC0zMzY1NTUxMQ%3D%3D
Comments: https://staging.kickstarter.com/projects/fjorden/fjorden-iphone-photography-reinvented/comments?comment=Q29tbWVudC0zMzYzMzAyOA%3D%3D&reply=Q29tbWVudC0zMzY0Nzg3Nw%3D%3D

- [x] Check while the app is backgrounded, terminated and in the foreground that a deeplink for both URLs responds appropriately.
- [x] Check while the app is backgrounded, terminated and in the foreground that a push notification for both URLs responds appropriately.

### CODE REVIEW

File: Kickstarter-iOS/ViewModels/AppDelegateViewModelTests.swift [hunk 1]
```diff
@@ -761,6 +783,28 @@ final class AppDelegateViewModelTests: TestCase {
...
+        "https://\(AppEnvironment.current.apiService.serverConfig.webBaseUrl.host ?? "")/projects/fjorden/fjorden-iphone-photography-reinvented/posts/3254626/comments?comment=Q29tbWVudC0zMzY0OTg0MQ%3D%3D&reply=deadbeef"
+
+      let result = self.vm.inputs.applicationOpenUrl(
+        application: UIApplication.shared,
+        url: URL(string: url)!,
+        options: [:]
+      )
+      XCTAssertTrue(result)
+
+      self.presentViewController.assertValues([4])
```
Comment (msadoon) [inline Kickstarter-iOS/ViewModels/AppDelegateViewModelTests.swift:804]: might be nice to know what is in the stack too. (ie `CommentRepliesViewController`) here and above.

File: Kickstarter-iOS/Views/Controllers/CommentRepliesViewController.swift [hunk 2]
```diff
@@ -114,6 +116,7 @@ final class CommentRepliesViewController: UITableViewController {
         guard let self = self else { return }
         self.dataSource.load(repliesAndTotalCount: repliesAndTotalCount, project: project)
         self.tableView.reloadData()
+        self.viewModel.inputs.loadRepliesAndProjectIntoDataSourceComplete()
```
Comment (msadoon) [inline Kickstarter-iOS/Views/Controllers/CommentRepliesViewController.swift:119]: could also call this input `dataSourceLoaded`

**Review by msadoon — APPROVED** [review]
good to go!

File: Kickstarter-iOS/ViewModels/AppDelegateViewModelTests.swift (hunk 1 above)
Comment (singhhari) [reply Kickstarter-iOS/ViewModels/AppDelegateViewModelTests.swift:804]: Ah so the view model output is being mapped to the observer as a count of view controllers so I think this is the designed testing behavior.

**Pull request merged**